plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/handler/DeleteResourceHandler.java

```diff
@@ -48,16 +49,24 @@ public Object execute(ExecutionEvent event) throws ExecutionException {
 			return OpenShiftUIActivator.statusFactory().cancelStatus("No resource selected that we can delete."); //$NON-NLS-1$
 		}
 		try (Stream<IResourceUIModel> stream = Arrays.stream(resources)) {
-	        boolean hasProject = stream.anyMatch(resource -> resource instanceof IProjectAdapter);
+	        boolean willDeleteSubResources = stream.anyMatch(resource -> Platform.getAdapterManager().getAdapter(resource.getResource(), ResourceProcessor.class).willCascadeDeleteLinkedResources());
+	        boolean confirm;
+	        
+	        String message = (resources.length > 1)?(willDeleteSubResources)?OpenShiftUIMessages.ResourceWithLinkedResourcesDeletionConfirmationN:OpenShiftUIMessages.ResourceDeletionConfirmationN
+                    :(willDeleteSubResources)?NLS.bind(OpenShiftUIMessages.ResourceWithLinkedResourcesDeletionConfirmation, resources[0].getResource().getName(), resources[0].getResource().getKind()):
+                                  NLS.bind(OpenShiftUIMessages.ResourceDeletionConfirmation, resources[0].getResource().getName(), resources[0].getResource().getKind());
+            if (willDeleteSubResources) {
+	            int status = new CheckboxMessageDialog(HandlerUtil.getActiveShell(event), OpenShiftUIMessages.ResourceDeletionDialogTitle, message, "Cascade delete").open();
+	            confirm = ((status & 0x01) == 0);
+	            willDeleteSubResources = (status & 0x02) == 2;
```

Comment:
A minor suggestion, please replace 0x02 with CheckboxMessageDialog.CHECKBOX_SELECTED and 0x01 with Window.CANCEL.